### DISCUSSION TITLE
Question about supervisor.c

### DISCUSSION THREAD

**Discussion by wdliu356** [bitcraze/discussions#1609] [Q&A] [answered]
Hi
I'm reading the source code and feel bit confused about the `isTumbledCheck()` function.
In this function, whether the drone is tilted is defined as `const bool isTilted = (data->acc.z < acceptedTiltAccZ);`, which make no sense to me. I don't understand why we are using the acceleration along z axis to make the decision. Shouldn't that be something related to angular velocity or angle reading?
Best

**Comment by gemenerik** [comment, marked as answer]
Using an accelerometer for orientation checks is effective as it measures both the forces acting on the drone—such as those from flying or maneuvers—and the acceleration due to gravity.

For example, when the drone is perfectly upright and not accelerating, the accelerometer will read a positive value reflecting the force of gravity. If you tilt the drone or flip it upside down, the z-axis reading will change accordingly, indicating the new orientation. This capability makes accelerometers particularly reliable for determining tilt without the drift that can affect gyroscopic measurements over time.

While you could theoretically accelerate upwards quickly enough to trigger tumble detection, it would require exceeding the drone's maximum thrust.

It can be fun to add a logging configuration in the cfclient for the accelerometer's z-axis (acc.z). This allows you to plot the data in the plotter tab and gain insights into how the acceleration relates to drone orientation during flight or while holding it in your hand.
